app/src/main/java/com/medmapper/v33001/dto/Medicine.kt

```diff
@@ -6,10 +6,12 @@ import androidx.room.PrimaryKey
 
 @Entity(tableName = "Medicine")
 data class Medicine(
-    @PrimaryKey(autoGenerate = true) val medID: Int,
-    @ColumnInfo(name = "name") val name: String?,
+    @PrimaryKey(autoGenerate = true) var medID: Int,
+    @ColumnInfo(name = "name") val name: String,
     @ColumnInfo(name = "strength") val strength: String?,
     @ColumnInfo(name = "start date") val startDate: String?,
-    @ColumnInfo(name = "prescription length") val lengthInDays: Int
+    @ColumnInfo(name = "prescription length") val lengthInDays: Int?,
+    @ColumnInfo(name = "frequency") val frequency: String?,
+    @ColumnInfo(name = "reason") val reason: String?
```

Comment:
Added Frequency and Reason columns -> This could be useful for people who take medications twice a day, evening or night, or as needed, etc. Added Reason so this could be used as a guide when people talk to their doctors and not have to keep track of why they were prescribed what.

Reply:
This looks like an incomplete feature.
Are you using these anywhere? 

Recommendation: if not used, remove it, or keep it in a separate development branch until it is complete.  Keep the codebase as small as possible to do the job you want to do.  Unused code is clutter, and bad documentation, which increases technical debt.
Only complete features, which meet the team's Definition of Done, should be merged to Main/Master.